torch/_dynamo/utils.py

```diff
@@ -2126,3 +2127,13 @@ def get_static_address_type(t):
         return getattr(t, "_dynamo_static_input_type", None)
 
     return None
+
+
+def reset_rng_state(use_xla=False):
```

Comment:
nit: would be nice for this to be in the testing namespace because we shouldnt be using this otherwise

Reply:
yea, I also thought about this. One thing makes me hesitate is, `reset_rng_state` is also used in the benchmarking script (`benchmarks/dynamo/common.py`). It looks a bit weird to me to import something from testing package/namespace in benchmarks/dynamo/common.py . So I end up putting it in utils.py

Reply:
You can put it in torch.testing somewhere

Reply:
Maybe ?

Reply:
> You can put it in torch.testing somewhere

Is 'torch/testing/_internal/common_utils.py' a good place then?

Reply:
hmm, CI shows a lot of errors like
```
loading model: 0it [00:00, ?it/s]
Traceback (most recent call last):
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/torchbench.py", line 510, in <module>
    torchbench_main()
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/torchbench.py", line 506, in torchbench_main
    main(TorchBenchmarkRunner(), original_dir)
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/common.py", line 3011, in main
    process_entry(0, runner, original_dir, args)
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/common.py", line 2969, in process_entry
    return maybe_fresh_cache(
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/common.py", line 1634, in inner
    return fn(*args, **kwargs)
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/common.py", line 3428, in run
    ) = runner.load_model(
  File "/home/shunting/ws/pytorch/benchmarks/dynamo/torchbench.py", line 383, in load_model
    benchmark = benchmark_cls(
  File "/home/shunting/ws/pytorch/torchbenchmark/torchbenchmark/util/model.py", line 25, in __call__
    obj = type.__call__(cls, *args, **kwargs)
  File "/home/shunting/ws/pytorch/torchbenchmark/torchbenchmark/models/resnet18/__init__.py", line 11, in __init__
    super().__init__(model_name="resnet18", test=test, device=device,
  File "/home/shunting/ws/pytorch/torchbenchmark/torchbenchmark/util/framework/vision/model_factory.py", line 23, in __init__
    torch.backends.cudnn.deterministic = False
  File "/home/shunting/ws/pytorch/torch/backends/__init__.py", line 43, in __set__
    raise RuntimeError(
RuntimeError: not allowed to set torch.backends.cudnn flags after disable_global_flags; please use flags() context manager instead
```

after I move `reset_rng_state` to `torch/testing/_internal/common_utils.py`.  Don't know how they relate yet

Reply:
ok, here is the reason of the failures.

When importing 'torch.testing._internal.common_utils', we run `torch.backends.disable_global_flags()`. This will work like freezing the configs and make those config only be updated by specific APIs. But later on when we loading a TB model (maybe also apply to other benchmark suite), we change a cudnn config directly like `torch.backends.cudnn.deterministic` which cause the failure.

I think it does not worth changing TB code for this. Instead, we should not import `torch.testing._internal.common_utils` in our benchmark scripts because of the side effect of calling 'disable_global_flags()' . I'll revert this part of change and move `reset_rng_state` back to the old place

Reply:
Or, I'll move `reset_rng_state` to `torch._dynamo.testing` . Should be safe this time since `benchmarks.dynamo.common` already imports `torch._dynamo.testing`